OpenRA.Mods.Common/Activities/EnterTransport.cs

```diff
@@ -63,10 +63,6 @@ protected override void OnEnterComplete(Actor self, Actor targetActor)
 
 				enterCargo.Load(enterActor, self);
 				w.Remove(self);
-
-				// Preemptively cancel any activities to avoid an edge-case where successively queued
-				// EnterTransports corrupt the actor state. Activities are cancelled again on unload
-				self.CancelActivity();
```

Comment:
We will still need this to cancel any activities queued after the `EnterTransport` (e.g. backup orders to enter a different transport if the first one was full).

Reply:
...err maybe not? It would help if I read the comment above before making my own comments. Still, we should make sure to test that this still behaves correctly.

Reply:
This `CancelActivity` is not needed because we already cancel the activity queue in `UnloadCargo`. Cancelling the activity queue twice smells of a hack. It does nothing and if it did it would be a bug.